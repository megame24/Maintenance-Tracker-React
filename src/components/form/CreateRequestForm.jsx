import React from 'react';
import { PropTypes } from 'prop-types';
import { Dimmer, Loader } from 'semantic-ui-react';
import { Redirect } from 'react-router-dom';

const CreateRequestForm = (props) => {
  const {
    handleSubmit, title, description, type,
    handleChange, isLoading, errors, success
  } = props;
  return (
    <section id="make-request" className="min-height">
      <Dimmer
        page
        active={isLoading}
      >
        <Loader size="medium" />
      </Dimmer>
      <div className="container">
        {
          errors.message
          && (
            <div className="message error-message">
              {errors.message}
            </div>
          )
        }
        <div className="make-edit-request">
          <h2>Make a new request</h2>
          <hr />
          <form id="make-request-form" onSubmit={handleSubmit}>
            <label htmlFor="title"> Request title</label>
            <input
              className="form-input"
              type="text"
              id="title"
              name="title"
              onChange={handleChange}
              value={title}
              placeholder="E.g. Broken side mirror"
              required
            />
            <label htmlFor="description">Request Description</label>
            <textarea
              className="form-input"
              type="text"
              id="description"
              name="description"
              value={description}
              onChange={handleChange}
              required
            />
            <label htmlFor="type">Request type</label>
            <span>
              <input
                type="radio"
                name="type"
                onClick={handleChange}
                value="maintenance"
                defaultChecked={type === 'maintenance'}
              />
              &nbsp;
              Maintenance
            </span>
            &nbsp; &nbsp;
            <span>
              <input
                type="radio"
                name="type"
                onClick={handleChange}
                value="repair"
                defaultChecked={type === 'repair'}
              />
              &nbsp;
              Repair
            </span>
            <hr />
            <input
              type="submit"
              id="submit-btn"
              value="Make Request"
              className="form-input btn btn-primary"
            />
          </form>
        </div>
      </div>
      {
        success && <Redirect to="/view-requests" />
      }
    </section>
  );
};

CreateRequestForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  success: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

CreateRequestForm.defaultProps = {
  errors: {},
};

export default CreateRequestForm;
