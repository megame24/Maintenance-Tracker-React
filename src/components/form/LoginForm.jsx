import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

const LoginForm = (props) => {
  const {
    handleSubmit, formData, handleChange,
    isLoading, errors,
  } = props;
  return (
    <section id="login">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="form">
          <div className="form-icon-container">
            <div className="center form-icon blue">
              <i className="icon ion-person" />
            </div>
          </div>
          <h1 className="center">Create Account</h1>
          <form id="LoginForm" onSubmit={handleSubmit}>
            <input
              onChange={handleChange}
              value={formData.usernameOrEmail}
              className="form-input"
              type="text"
              name="usernameOrEmail"
              id="email"
              placeholder="Email or Username"
              required
            />
            <input
              onChange={handleChange}
              value={formData.password}
              className="form-input"
              type="password"
              name="password"
              id="password"
              placeholder="Password"
              required
            />
            <input
              className="form-input btn btn-primary"
              type="submit"
              id="submit-btn"
              value="Login"
            />
          </form>
          <div className="form-footer center">
            <p>
              Not a member? &nbsp;
              <Link to="/signup" className="blue">Create Account</Link>
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

LoginForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  formData: PropTypes.shape({
    password: PropTypes.string.isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

LoginForm.defaultProps = {
  errors: {},
};

export default LoginForm;
