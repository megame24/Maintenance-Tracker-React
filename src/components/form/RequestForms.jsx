import React from 'react';
import { PropTypes } from 'prop-types';
import { Redirect } from 'react-router-dom';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

export const RequestForm = ({
  handleSubmit, title, description, type,
  handleChange,
}) => (
  <form id="make-request-form" onSubmit={handleSubmit}>
    <label htmlFor="title"> Request title</label>
    <input
      className="form-input"
      type="text"
      id="title"
      name="title"
      onChange={handleChange}
      value={title}
      placeholder="E.g. Broken side mirror"
      required
    />
    <label htmlFor="description">Request Description</label>
    <textarea
      className="form-input"
      type="text"
      id="description"
      name="description"
      value={description}
      onChange={handleChange}
      required
    />
    <label htmlFor="type">Request type</label>
    <span>
      <input
        type="radio"
        name="type"
        onChange={handleChange}
        value="maintenance"
        checked={type === 'maintenance'}
      />
      &nbsp;
      Maintenance
    </span>
    &nbsp; &nbsp;
    <span>
      <input
        type="radio"
        name="type"
        onChange={handleChange}
        value="repair"
        checked={type === 'repair'}
      />
      &nbsp;
      Repair
    </span>
    <hr />
    <input
      type="submit"
      id="submit-btn"
      value="Make Request"
      className="form-input btn btn-primary"
    />
  </form>
);

RequestForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  type: PropTypes.string,
};

RequestForm.defaultProps = {
  title: '',
  description: '',
  type: 'repair',
};

export const CreateRequestForm = (props) => {
  const {
    handleSubmit, title, description, type,
    handleChange, isLoading, errors, success
  } = props;
  return (
    <section id="make-request" className="min-height">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="make-edit-request">
          <h2>Make a new request</h2>
          <hr />
          <RequestForm
            handleChange={handleChange}
            handleSubmit={handleSubmit}
            title={title}
            description={description}
            type={type}
          />
        </div>
      </div>
      {
        success && <Redirect to="/users/requests/all" />
      }
    </section>
  );
};

CreateRequestForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  success: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

CreateRequestForm.defaultProps = {
  errors: {},
};

export const EditRequestForm = (props) => {
  const {
    handleSubmit, title, description, type,
    handleChange, isLoading, errors, success
  } = props;
  return (
    <section id="make-request" className="min-height">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="make-edit-request">
          <h2>Make a new request</h2>
          <hr />
          <RequestForm
            handleChange={handleChange}
            handleSubmit={handleSubmit}
            title={title}
            description={description}
            type={type}
          />
        </div>
      </div>
      {
        success && <Redirect to="/users/requests/all" />
      }
    </section>
  );
};

EditRequestForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  type: PropTypes.string,
  success: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

EditRequestForm.defaultProps = {
  errors: {},
  title: '',
  description: '',
  type: 'repair',
};

export default {};
