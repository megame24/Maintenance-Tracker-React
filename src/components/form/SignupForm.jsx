import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import Loading from '../Loading';
import ErrorMessage from '../ErrorMessage';

const SignupForm = (props) => {
  const {
    handleSubmit, formData, handleChange,
    isLoading, errors,
  } = props;
  return (
    <section id="register">
      <Loading isLoading={isLoading} />
      <div className="container">
        <ErrorMessage errors={errors} />
        <div className="form">
          <div className="form-icon-container">
            <div className="center form-icon blue">
              <i className="icon ion-person" />
            </div>
          </div>
          <h1 className="center">Create Account</h1>
          <form id="signupForm" onSubmit={handleSubmit}>
            <input
              onChange={handleChange}
              value={formData.name}
              className="form-input"
              name="fullname"
              type="text"
              id="name"
              placeholder="Full Name"
              required
            />
            <input
              onChange={handleChange}
              value={formData.email}
              className="form-input"
              type="text"
              name="email"
              id="email"
              placeholder="Email"
              required
            />
            <input
              onChange={handleChange}
              value={formData.username}
              className="form-input"
              type="text"
              name="username"
              id="username"
              placeholder="Username"
              required
            />
            <input
              onChange={handleChange}
              value={formData.password}
              className="form-input"
              type="password"
              name="password"
              id="password"
              placeholder="Password"
              required
            />
            <input
              className="form-input btn btn-primary"
              type="submit"
              id="submit-btn"
              value="Create Account"
            />
          </form>
          <div className="form-footer center">
            <p>
              Already a member? &nbsp;
              <Link to="/login" className="blue">Login</Link>
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

SignupForm.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  formData: PropTypes.shape({
    email: PropTypes.string.isRequired
  }).isRequired,
  isLoading: PropTypes.bool.isRequired,
  errors: PropTypes.shape({
    statusCode: PropTypes.number
  }),
};

SignupForm.defaultProps = {
  errors: {},
};

export default SignupForm;
