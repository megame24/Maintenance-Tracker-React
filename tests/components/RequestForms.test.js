import React from 'react';
import { shallow } from 'enzyme';
import { CreateRequestForm, RequestForm, EditRequestForm } from '../../src/components/form/RequestForms';

const props = {
  handleChange: () => {},
  handleSubmit: () => {},
  title: 'title',
  description: 'des',
  type: 'type',
  success: false,
  isLoading: false,
  errors: {},
}

describe('Testing RequestForm component', () => {
  it('should render as expected', () => {
    const tree = shallow(<RequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
});

describe('testing CreateRequestForm component', () => {
  it('should render as expected when there is no error', () => {
    const tree = shallow(<CreateRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when there is an error', () => {
    props.errors = {
      message: 'error'
    };
    const tree = shallow(<CreateRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when success is true', () => {
    props.success = true;
    const tree = shallow(<CreateRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
});

describe('testing EditRequestForm component', () => {
  const props = {
    handleChange: () => {},
    handleSubmit: () => {},
    title: 'title',
    description: 'des',
    type: 'type',
    success: false,
    isLoading: false,
    errors: {},
  }
  it('should render as expected when there is no error', () => {
    const tree = shallow(<EditRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when there is an error', () => {
    props.errors = {
      message: 'error'
    };
    const tree = shallow(<EditRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when success is true', () => {
    props.success = true;
    const tree = shallow(<EditRequestForm {...props} />)
    expect(tree).toMatchSnapshot();
  });
});

