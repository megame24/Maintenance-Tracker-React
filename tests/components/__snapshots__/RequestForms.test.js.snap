// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RequestForm component should render as expected 1`] = `
<form
  id="make-request-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="title"
  >
     Request title
  </label>
  <input
    className="form-input"
    id="title"
    name="title"
    onChange={[Function]}
    placeholder="E.g. Broken side mirror"
    required={true}
    type="text"
    value="title"
  />
  <label
    htmlFor="description"
  >
    Request Description
  </label>
  <textarea
    className="form-input"
    id="description"
    name="description"
    onChange={[Function]}
    required={true}
    type="text"
    value="des"
  />
  <label
    htmlFor="type"
  >
    Request type
  </label>
  <span>
    <input
      checked={false}
      name="type"
      onChange={[Function]}
      type="radio"
      value="maintenance"
    />
      Maintenance
  </span>
     
  <span>
    <input
      checked={false}
      name="type"
      onChange={[Function]}
      type="radio"
      value="repair"
    />
      Repair
  </span>
  <hr />
  <input
    className="form-input btn btn-primary"
    id="submit-btn"
    type="submit"
    value="Make Request"
  />
</form>
`;

exports[`testing CreateRequestForm component should render as expected when success is true 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={
        Object {
          "message": "error",
        }
      }
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="create"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
  <Redirect
    push={false}
    to="/users/requests/all"
  />
</section>
`;

exports[`testing CreateRequestForm component should render as expected when there is an error 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={
        Object {
          "message": "error",
        }
      }
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="create"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
</section>
`;

exports[`testing CreateRequestForm component should render as expected when there is no error 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={Object {}}
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="create"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
</section>
`;

exports[`testing EditRequestForm component should render as expected when success is true 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={
        Object {
          "message": "error",
        }
      }
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="edit"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
  <Redirect
    push={false}
    to="/users/requests/all"
  />
</section>
`;

exports[`testing EditRequestForm component should render as expected when there is an error 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={
        Object {
          "message": "error",
        }
      }
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="edit"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
</section>
`;

exports[`testing EditRequestForm component should render as expected when there is no error 1`] = `
<section
  className="min-height"
  id="make-request"
>
  <Loading
    isLoading={false}
  />
  <div
    className="container"
  >
    <ErrorMessage
      errors={Object {}}
    />
    <div
      className="make-edit-request"
    >
      <h2>
        Make a new request
      </h2>
      <hr />
      <RequestForm
        description="des"
        formType="edit"
        handleChange={[Function]}
        handleSubmit={[Function]}
        title="title"
        type="type"
      />
    </div>
  </div>
</section>
`;
